shell/browser/win/dark_mode.cc

```diff
@@ -1,178 +1,51 @@
-// Copyright (c) 2020 Microsoft Inc. All rights reserved.
+// Copyright (c) 2022 Microsoft Inc. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE-CHROMIUM file.
 
 #include "shell/browser/win/dark_mode.h"
 
 #include <dwmapi.h>  // DwmSetWindowAttribute()
 
-#include "base/files/file_path.h"
-#include "base/scoped_native_library.h"
-#include "base/win/pe_image.h"
-#include "base/win/win_util.h"
-#include "base/win/windows_version.h"
+#define DARK_MODE_STRING_NAME L"DarkMode_Explorer"
+#define DWMWA_USE_IMMERSIVE_DARK_MODE_BEFORE_20H1 19
+#define DWMWA_USE_IMMERSIVE_DARK_MODE 20
 
 // This namespace contains code originally from
-// https://github.com/ysc3839/win32-darkmode/
-// governed by the MIT license and (c) Richard Yu
+// https://github.com/microsoft/terminal
+// governed by the MIT license and (c) Microsoft Corporation.
 namespace {
 
-// 1903 18362
-enum PreferredAppMode { Default, AllowDark, ForceDark, ForceLight, Max };
+// Use private Windows API to set window theme.
+HRESULT TrySetWindowTheme(HWND hWnd, bool dark) {
+  HRESULT result =
+      SetWindowTheme(hWnd, dark ? DARK_MODE_STRING_NAME : L"", nullptr);
```

Comment:
Interestingly in my testing on an up-to-date Windows 10 machine, this wasn't necessary, and nothing changed if I removed it.

No suggestion or anything, just making a note.

Reply:
Hmm that's weird, no effect for me on Win11 either. I believe it had something to do with lists or menus - but it also might've been for an older Win10 version. If we remove it, we would get rid of the last "magic value" being used.

Reply:
Even with it, scrollbars, buttons, and multiple `selects` are light mode styling unless you set `:root { color-scheme: light dark; }` in the CSS for the page, at which point they style as expected. I've got a note to do a PR updating the dark mode tutorial to add that CSS, as the current tutorial renders with light mode buttons since it doesn't have that.

So I guess the only question would be if it's serving a purpose on older Win10, say 20H1 and older. I'll grab a VM to test that. If it seems to have no effect there either, then I vote we just drop it since it doesn't get us anything (that we know of). If it is needed on older Win10, I vote we either wrap it in an if guard, or just don't support dark mode for those older versions.

Reply:
I tested it on Windows 1903, which is the first build where dark mode functionality exists. Does appear that `L"DarkMode_Explorer"` makes no difference, it's not necessary to get the titlebar black like we want.

So I vote we drop it, put only Win10 behind the fuse, and only do the toggle titlebar code on Win10. 🙂

Reply:
ok, yeah if we don't see any difference, let's remove it and if ever an issue comes up, we can try adding it again